lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -151,14 +151,55 @@ function makeComparison(name, actualResult, expectedResult) {
   };
 }
 
+/**
+ * Delete expectations that don't match environment criteria.
+ * @param {LocalConsole} localConsole
+ * @param {LH.Result} lhr
+ * @param {Smokehouse.ExpectedRunnerResult} expected
+ */
+function pruneExpectations(localConsole, lhr, expected) {
+  const userAgent = lhr.userAgent;
+  const userAgentMatch = /Chrome\/(\d+)/.exec(userAgent); // Chrome/85.0.4174.0
+  if (!userAgentMatch) throw new Error('Could not get chrome version.');
+  const actualChromeVersion = Number(userAgentMatch[1]);
+
+  /**
+   * @param {*} obj
+   */
+  function failsChromeVersionCheck(obj) {
+    if (!obj._minChromeMajorVersion) return false;
+    return actualChromeVersion < obj._minChromeMajorVersion;
+  }
+
+  /**
+   * @param {*} obj
+   */
+  function pruneNewerChromeExpectations(obj) {
+    for (const key of Object.keys(obj)) {
+      const value = obj[key];
+      if (!value || typeof value !== 'object') continue;
+      else if (failsChromeVersionCheck(value)) {
+        localConsole.log(`[${key}] failed chrome version check, pruning expectation: ${
+          JSON.stringify(value, null, 2)}`);
+        delete obj[key];
+      } else pruneNewerChromeExpectations(value);
+    }
+    delete obj._minChromeMajorVersion;
+  }
+
+  pruneNewerChromeExpectations(expected);
+}
+
 /**
  * Collate results into comparisons of actual and expected scores on each audit/artifact.
  * @param {LocalConsole} localConsole
- * @param {Smokehouse.ExpectedRunnerResult} actual
+ * @param {{lhr: LH.Result, artifacts: LH.Artifacts}} actual
  * @param {Smokehouse.ExpectedRunnerResult} expected
  * @return {Comparison[]}
  */
 function collateResults(localConsole, actual, expected) {
+  pruneExpectations(localConsole, actual.lhr, expected);
```

Comment:
this should probably go down in `report()` so obvious upon entry into the file, not down deep

Reply:
i like moving it too. felt a bit buried.